### ISSUE TITLE
Helm Chart - apiVersion not set - k8s version 1.10+

### ISSUE THREAD

**Issue by lnxmac** [kubernetes/autoscaler#5364]
**Which component are you using?**: cluster-autoscaler

**What version of the component are you using?**: Helm 9.21.1 cluster-autoscaler

Component version:

**What k8s version are you using (`kubectl version`)?**: 1.22

**What did you expect to happen?**: Upgrade 9.21.0 -> 9.21.1 to work

**What happened instead?**:
Helm upgrade fails: 

Error: error validating "": error validating data: apiVersion not set

**How to reproduce it (as minimally and precisely as possible)**:

Helm upgrade from 9.21.0 to 9.21.1 on a k8s version 1.10+

**Anything else we need to know?**:

There seems so be a problem in the if/else construct, it seems the 2nd if is nested, so it never checks for >1.21-0 and apiVersion is not set.
**charts/cluster-autoscaler/templates/_helpers.tpl** -> 
```{{/*
Return the appropriate apiVersion for podsecuritypolicy.
*/}}
{{- define "podsecuritypolicy.apiVersion" -}}
{{- $kubeTargetVersion := default .Capabilities.KubeVersion.GitVersion .Values.kubeTargetVersionOverride }}
{{- if semverCompare "<1.10-0" $kubeTargetVersion -}}
{{- print "extensions/v1beta1" -}}
{{- if semverCompare ">1.21-0" $kubeTargetVersion -}}
{{- print "policy/v1" -}}
{{- else -}}
{{- print "policy/v1beta1" -}}
{{- end -}}
{{- end -}}
{{- end -}}```

**Comment by jim-barber-he** [comment]
Since this issue is nearly 24 hours old I thought I'd create a pull request to save the authors of this chart having to deal with it.
I was going to attempt to fix the above like so:
```diff
diff --git a/charts/cluster-autoscaler/templates/_helpers.tpl b/charts/cluster-autoscaler/templates/_helpers.tpl
index 3fbf00965..25d1eb824 100644
--- a/charts/cluster-autoscaler/templates/_helpers.tpl
+++ b/charts/cluster-autoscaler/templates/_helpers.tpl
@@ -70,13 +70,12 @@ Return the appropriate apiVersion for podsecuritypolicy.
 {{- $kubeTargetVersion := default .Capabilities.KubeVersion.GitVersion .Values.kubeTargetVersionOverride }}
 {{- if semverCompare "<1.10-0" $kubeTargetVersion -}}
 {{- print "extensions/v1beta1" -}}
-{{- if semverCompare ">1.21-0" $kubeTargetVersion -}}
+{{- else if semverCompare ">1.21-0" $kubeTargetVersion -}}
 {{- print "policy/v1" -}}
 {{- else -}}
 {{- print "policy/v1beta1" -}}
 {{- end -}}
 {{- end -}}
-{{- end -}}
 
 {{/*
 Return the appropriate apiVersion for podDisruptionBudget.
```

However this doesn't work with Helm outputting this error:
```
  Error: Failed to render chart: exit status 1: Error: unable to build kubernetes objects from release manifest: resource mapping not found for name: "cluster-autoscaler" namespace: "" from "": no matches for kind "PodSecurityPolicy" in version "policy/v1"
  ensure CRDs are installed first
```

This is  because in a 1.24 cluster the `apiVersion` for `PodSecurityPolicy` is still `policy/v1beta1`.
```
$ kubectl version -o json | jq -r .serverVersion.gitVersion  
v1.24.8
```
```
$ kubectl explain PodSecurityPolicy | head -2
KIND:     PodSecurityPolicy
VERSION:  policy/v1beta1
```

Then in Kubernetes v1.25 PodSecurityPolicies are removed completely: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#psp-v125

So it seems this chart needs more extensive changes to support clusters from version 1.25 onwards to replace `PodSecurityPolicy` with the new mechanisms or to remove them?

**Comment by jim-barber-he** [comment]
It seems like the easiest course of action is to revert the change that version `9.21.1` of the Helm chart introduced, then anyone with a 1.25 or newer cluster needs to make sure that `rbac.pspEnabled` remains set to its default value of `false`.

**Comment by marianobilli** [comment]
Same, so basically they will not fix it and once we have kuberentes 1.25 it will not be a "problem" anymore ? :P
